Sources/Datadog/DatadogConfiguration.swift

```diff
@@ -127,6 +129,17 @@ extension Datadog {
                 return self
             }
 
+            /// Sets the hosts to be automatically traced.
+            ///
+            /// Example, if tracedHosts is ["https://foo.bar"], then every network request such as "https://foo.bar/some/path" will be automatically traced.
+            /// If nil, automatic tracing is disabled.
+            ///
+            /// - Parameter tracedHosts: `nil` by default
```

Comment:
This public API comment needs a fix - the `https://foo.bar` is not a valid input. Let's use a realistic example, like `example.com` + `api.example.com` instead of `foo.bar`.

It also doesn't mention `URLSession`, nor distributed tracing and doesn't explain what "automatically traced" means. IMO the [comment in Android SDK](https://github.com/DataDog/dd-sdk-android/blob/34c6d35ccf4c1f88fb64df47861ffbed5dc06ca8/dd-sdk-android/src/main/kotlin/com/datadog/android/tracing/TracingInterceptor.kt#L16-L39) is a good reference, let's elevate it to that level (?) 📈.

Reply:
IMO `tracedHost: example.com` including `api.example.com` is already very confusing in iOS as those URLs have different `url.host`s. we can follow Android and not mention this behavior in comments, wdyt?

```swift
Sets the hosts to be automatically traced.

Every request made to a traced host will create its Span with related information; such as url, method, status code, error (if any).

Example, if tracedHosts is ["https://example.com"], then every network request such as "https://example.com/some/path" will be automatically traced and generate a span.
If tracedHosts is empty, automatic tracing is disabled.

IMPORTANT: Non-empty tracedHosts modifies implementation of URLSession methods, in case your app relies on URLSession internals please refer to URLSessionSwizzler.swift file
```
i'm not so sure of mentioning distributed tracing as we support it only in some cases. does that sound good?

Reply:
> IMO tracedHost: example.com including api.example.com is already very confusing in iOS as those URLs have different url.hosts. we can follow Android and not mention this behavior in comments, wdyt?

We want the same behaviour as Android SDK, which is:
```python
if tracedHosts == ["example.com"], then:

https://api.example.com/v1/oauth     - traced
https://example.com/api/v1/oauth     - traced
http://www.example.com/api/v1/oauth  - traced
https://foo.com/                     - not traced


if tracedHosts == ["api.example.com"], then:

https://api.example.com/v1/oauth     - traced
https://example.com/api/v1/oauth     - not traced
http://www.example.com/api/v1/oauth  - not traced
https://foo.com/                     - not traced
```

If we expect the user to provide schema (`https`/`http`), then this concept gets broken or becomes non-obvious, no?

> i'm not so sure of mentioning distributed tracing as we support it only in some cases. does that sound good?

IMO we should mention it, as it's a powerful integration. If it doesn't work for everyone, we can say _"If your backend is instrumented with Datadog, ..."_.

Reply:
oh yes, i'm removing `https` from comments 👍 

i mean it doesn't work for `dataTaskWithURL` in iOS 13, shall we mention this feature anyway?